cogs/onmyoji.py

```diff
@@ -285,6 +286,153 @@ def shiki_bounty_embed(self, shiki):
+        """
+        print(self.shard_trading_db)
+        await ctx.send(f"Shard Lists for <@{str(ctx.message.author.id)}>:")
+        try:
+            need_list = '\n'.join(self.shard_trading_db[str(ctx.message.author.id)]['need'])
+            await ctx.send(f"Shards you **Need**: ```\n{need_list}```")
+            print(need_list)
+        except KeyError:
+            await ctx.send("You have nothing in your Needed list yet.")
+        try:
+            need_list = '\n'.join(self.shard_trading_db[str(ctx.message.author.id)]['have'])
+            await ctx.send(f"Shards you **Have**: ```\n{have_list}```")
+        except KeyError:
+            await ctx.send("You have nothing in your Have list yet.")
+
+    @shard.command(name="need")
+    async def shard_set_need(self,ctx,*,args):
+        """
+        Returns the list of shards you need.
+        If shards are listed, assigns those shards to your needed list.
+        """
+        arg_list = args.split("\n")
+        self.shard_load_json()
+        try: 
+            self.shard_trading_db[str(ctx.message.author.id)]['need'] = arg_list
+        except KeyError:
+            self.shard_trading_db[str(ctx.message.author.id)] = {}
+            self.shard_trading_db[str(ctx.message.author.id)]['need'] = arg_list
+        self.shard_file_writeout()
+        arg_string = '\n'.join(self.shard_trading_db[str(ctx.message.author.id)]['need'])
+        await ctx.send(f'The shards you need are now set to: ```\n{arg_string}```')
+
+    @shard.command(name="have")
+    async def shard_set_have(self,ctx,*,args):
+        """
+        Returns the list of shards you have.
+        If shards are listed, assigns those shards to your have list.
+        """
+        arg_list = args.split("\n")
+        self.shard_load_json()
+        try: 
+            self.shard_trading_db[str(ctx.message.author.id)]['have'] = arg_list
+        except KeyError:
+            self.shard_trading_db[str(ctx.message.author.id)] = {}
+            self.shard_trading_db[str(ctx.message.author.id)]['have'] = arg_list
+        self.shard_file_writeout()
+        arg_string = '\n'.join(self.shard_trading_db[str(ctx.message.author.id)]['have'])
+        await ctx.send(f'The shards you have are now set to: ```\n{arg_string}```')
+
+    def compare_shard_db(main_user, other_user):
+        user1_have_list = [''.join(i for i in value if not i.isdigit()) for value in self.shard_trading_db[main_user]['have']]
+        user1_have_list = [i.strip() for i in user1_have_list]
+        user1_need_list = [value for value in self.shard_trading_db[main_user]['need']]
+        user2_have_list = [''.join(i for i in value if not i.isdigit()) for value in self.shard_trading_db[str(other_user)]['have']]
+        user2_have_list = [i.strip() for i in user2_have_list]
+        user2_need_list = [value for value in self.shard_trading_db[other_user]['need']]
```

Comment:
The fact that both `*_have_list` is a good sign you want to functionalize (DRY - don't repeat yourself)

Reply:
how do you mean